Source/FixedBackground/FixedBGEnergyAndMomFlux.hpp

```diff
@@ -0,0 +1,131 @@
+#define FIXEDBGENERGYANDMOMFLUX_HPP_
+
+#include "ADMFixedBGVars.hpp"
+#include "CCZ4Geometry.hpp"
+#include "Cell.hpp"
+#include "CoordinateTransformations.hpp"
+#include "Coordinates.hpp"
+#include "FourthOrderDerivatives.hpp"
+#include "GRInterval.hpp"
+#include "Tensor.hpp"
+#include "UserVariables.hpp" //This files needs NUM_VARS - total number of components
+#include "VarsTools.hpp"
+#include "simd.hpp"
+
+//! Calculates the momentum flux S_i with type matter_t and writes it to the
+//! grid
+template <class matter_t, class background_t> class FixedBGEnergyAndMomFlux
+{
+    // Use the variable definition in the matter class
+    template <class data_t>
+    using MatterVars = typename matter_t::template Vars<data_t>;
+
+    // Now the non grid ADM vars
+    template <class data_t> using MetricVars = ADMFixedBGVars::Vars<data_t>;
+
+  protected:
+    const FourthOrderDerivatives
+        m_deriv; //!< An object for calculating derivatives of the variables
+    const matter_t m_matter;                        //!< The matter object
+    const double m_dx;                              //!< The grid spacing
+    const background_t m_background;                //!< The metric background
+    const std::array<double, CH_SPACEDIM> m_center; //!< The grid center
+
+  public:
+    FixedBGEnergyAndMomFlux(matter_t a_matter, background_t a_background,
+                            double a_dx,
+                            std::array<double, CH_SPACEDIM> a_center)
+        : m_matter(a_matter), m_deriv(a_dx), m_dx(a_dx),
+          m_background(a_background), m_center(a_center)
+    {
+    }
+
+    template <class data_t> void compute(Cell<data_t> current_cell) const
+    {
+        // copy data from chombo gridpoint into local variables, and derivs
+        const auto vars = current_cell.template load_vars<MatterVars>();
+        const auto d1 = m_deriv.template diff1<MatterVars>(current_cell);
+
+        // get the metric vars from the background
+        Coordinates<data_t> coords(current_cell, m_dx, m_center);
+
+        MetricVars<data_t> metric_vars;
+        m_background.compute_metric_background(metric_vars, current_cell);
+
+        using namespace TensorAlgebra;
+        using namespace CoordinateTransformations;
+        //	const auto gamma = metric_vars.gamma;
+        const auto gamma_UU = compute_inverse_sym(metric_vars.gamma);
+        const auto chris_phys =
+            compute_christoffel(metric_vars.d1_gamma, gamma_UU);
+        const emtensor_t<data_t> emtensor = m_matter.compute_emtensor(
+            vars, metric_vars, d1, gamma_UU, chris_phys.ULL);
+        const auto lapse = metric_vars.lapse;
+        const auto shift = metric_vars.shift;
+        const data_t det_gamma = compute_determinant_sym(metric_vars.gamma);
+        Tensor<2, data_t> spherical_gamma = cartesian_to_spherical_LL(
+            metric_vars.gamma, coords.x, coords.y, coords.z);
+        data_t dArea = area_element_sphere(spherical_gamma);
```

Comment:
We can remove the need for this by not normalising the si_L and then multiplying the fluxes by det_gamma - see:
https://github.com/KAClough/GRChombo_public/blob/amelia/taggingcomparison_kcamend_refined_initial_data/Examples/Oscillaton/MatterEnergy.hpp